presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -525,24 +529,37 @@ protected Node visitShowFunctions(ShowFunctions node, Void context)
         {
             ImmutableList.Builder<Expression> rows = ImmutableList.builder();
             for (SqlFunction function : metadata.listFunctions(session)) {
-                rows.add(row(
-                        function.getSignature().getName().getFunctionNamespace().equals(DEFAULT_NAMESPACE) ?
-                                new StringLiteral(function.getSignature().getNameSuffix()) :
-                                new StringLiteral(function.getSignature().getName().toString()),
-                        new StringLiteral(function.getSignature().getReturnType().toString()),
-                        new StringLiteral(Joiner.on(", ").join(function.getSignature().getArgumentTypes())),
-                        new StringLiteral(getFunctionType(function)),
-                        function.isDeterministic() ? TRUE_LITERAL : FALSE_LITERAL,
-                        new StringLiteral(nullToEmpty(function.getDescription()))));
+                Signature signature = function.getSignature();
+                boolean builtIn = signature.getName().getFunctionNamespace().equals(DEFAULT_NAMESPACE);
+                ImmutableList.Builder<Expression> values = ImmutableList.<Expression>builder()
+                        .add(new StringLiteral(builtIn ? signature.getNameSuffix() : signature.getName().toString()))
+                        .add(new StringLiteral(signature.getReturnType().toString()))
+                        .add(new StringLiteral(Joiner.on(", ").join(signature.getArgumentTypes())))
+                        .add(new StringLiteral(getFunctionType(function)))
+                        .add(function.isDeterministic() ? TRUE_LITERAL : FALSE_LITERAL);
+                if (isShowBuiltInColumn(session)) {
+                    values.add(builtIn ? TRUE_LITERAL : FALSE_LITERAL);
+                }
+                if (isShowVariableArityInColumn(session)) {
+                    values.add(signature.isVariableArity() ? TRUE_LITERAL : FALSE_LITERAL);
+                }
+                values.add(new StringLiteral(nullToEmpty(function.getDescription())));
+                rows.add(new Row(values.build()));
             }
 
-            Map<String, String> columns = ImmutableMap.<String, String>builder()
+            ImmutableMap.Builder<String, String> columnsBuilder = ImmutableMap.<String, String>builder()
                     .put("function_name", "Function")
                     .put("return_type", "Return Type")
                     .put("argument_types", "Argument Types")
                     .put("function_type", "Function Type")
-                    .put("deterministic", "Deterministic")
-                    .put("description", "Description")
+                    .put("deterministic", "Deterministic");
+            if (isShowBuiltInColumn(session)) {
```

Comment:
Put the new columns at the end to minimize the possibility of breaking things. I don't think it's necessary to add config property and session property for these. Just add the information to `SHOW FUNCTION` is enough.